src/legacy/core_plugins/kibana/public/management/sections/settings/components/component_registry.ts

```diff
@@ -45,14 +48,20 @@ export function tryRegisterSettingsComponent(id, component) {
  * @param {*} component the component
  * @param {*} allowOverride (default: false) - optional flag to allow this component to override a previously registered component
  */
-export function registerSettingsComponent(id, component, allowOverride = false) {
+export function registerSettingsComponent(
+  id: Id,
+  component: FunctionComponent,
+  allowOverride = false
+) {
   if (!allowOverride && id in registry) {
     throw new Error(`Component with id ${id} is already registered.`);
   }
 
   // Setting a display name if one does not already exist.
   // This enhances the snapshots, as well as the debugging experience.
+  // @ts-ignore
   if (!component.displayName) {
+    // @ts-ignore
     component.displayName = id;
```

Comment:
Instead of disabling TypeScript compiler for these lines, is it possible to cast here?

```ts
if (!(component as any).displayName) {
```

Reply:
Why is this actually needed ? The property seems present

```typescript
    interface FunctionComponent<P = {}> {
        (props: PropsWithChildren<P>, context?: any): ReactElement | null;
        propTypes?: WeakValidationMap<P>;
        contextTypes?: ValidationMap<any>;
        defaultProps?: Partial<P>;
        displayName?: string;
    }
```

Reply:
I just removed this code as its just for debugging and it was giving me some trouble.